cpp/client.cc

```diff
@@ -0,0 +1,324 @@
+      }
+    }
+    Status status = reader->Finish();
+    if (!status.ok()) {
+      std::cout << status.error_code() << ": " << status.error_message()
+                << std::endl;
+    }
+  }
+
+ private:
+  std::unique_ptr<Wallet::Stub> stub_;
+};
+
+class StatsClient {
+ public:
+  StatsClient(std::shared_ptr<Channel> channel)
+      : stub_(Stats::NewStub(channel)) {}
+
+  void FetchPrice(const std::string& user) {
+    PriceRequest request;
+    PriceResponse response;
+    ClientContext context;
+    if (user == "Alice") {
+      context.AddMetadata("authorization", "2bd806c9");
+      context.AddMetadata("membership", "premium");
+    } else {
+      context.AddMetadata("authorization", "81b637d8");
+      context.AddMetadata("membership", "normal");
+    }
+    Status status = stub_->FetchPrice(&context, request, &response);
+    if (status.ok()) {
+      auto metadata_hostname =
+          context.GetServerInitialMetadata().find("hostname");
+      if (metadata_hostname != context.GetServerInitialMetadata().end()) {
+        std::cout << "server host: "
+                  << std::string(metadata_hostname->second.data(),
+                                 metadata_hostname->second.length())
+                  << std::endl;
+      }
+      std::cout << "grpc-coin price: " << response.price() << std::endl;
+    } else {
+      std::cout << status.error_code() << ": " << status.error_message()
+                << std::endl;
+    }
+  }
+
+  void WatchPrice(const std::string& user) {
+    PriceRequest request;
+    PriceResponse response;
+    ClientContext context;
+    if (user == "Alice") {
+      context.AddMetadata("authorization", "2bd806c9");
+      context.AddMetadata("membership", "premium");
+    } else {
+      context.AddMetadata("authorization", "81b637d8");
+      context.AddMetadata("membership", "normal");
+    }
+    std::unique_ptr<ClientReader<PriceResponse> > reader(
+        stub_->WatchPrice(&context, request));
+    while (reader->Read(&response)) {
+      auto metadata_hostname =
+          context.GetServerInitialMetadata().find("hostname");
+      if (metadata_hostname != context.GetServerInitialMetadata().end()) {
+        std::cout << "server host: "
+                  << std::string(metadata_hostname->second.data(),
+                                 metadata_hostname->second.length())
+                  << std::endl;
+      }
+      std::cout << "grpc-coin price: " << response.price() << std::endl;
+    }
+    Status status = reader->Finish();
+    if (!status.ok()) {
+      std::cout << status.error_code() << ": " << status.error_message()
+                << std::endl;
+    }
+  }
+
+ private:
+  std::unique_ptr<Stats::Stub> stub_;
+};
+
+int main(int argc, char** argv) {
+  std::string command = "balance";
+  std::string wallet_server = "localhost:50051";
+  std::string stats_server = "localhost:50052";
+  std::string user = "Alice";
+  bool watch = false;
+  bool unary_watch = false;
```

Comment:
`unary_watch` is never read.

Reply:
ah, i did miss this functionality, thanks for catching!